com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -751,13 +751,18 @@ private void OnNetworkPreUpdate()
                 }
             }
 
-            if (NetworkConfig.NetworkTransport is ITransportProfilerData profileTransport)
+            var data = PerformanceDataManager.GetData();
+            var eventHandler = OnPerformanceDataEvent;
+            if (eventHandler != null && data!= null)
```

Comment:
I guess you could warn and say "Did you forget to call RegisterMLAPIPerformanceEvent() first?"